rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java

```diff
@@ -222,6 +221,16 @@ private void addEmbeddedProject(HalResource halResource, Project project) {
         halResource.addEmbeddedResource("sw360:projects", halProject);
     }
 
+    public Component updateComponent(Component componentToUpdate, Component requestBodyComponent) {
+        for(Component._Fields field:Component._Fields.values()) {
+            Object fieldValue = requestBodyComponent.getFieldValue(field);
+            if(fieldValue != null) {
+                componentToUpdate.setFieldValue(field, fieldValue);
```

Comment:
This will allow to overwrite all component fields especially name, id..
Should we restrict some fields here?

Also there is no validation for some fields like links to another resources (e.g. users).
I think a validation is not necessary here but we should keep that in mind.

Reply:
If the request body contains a new `id` the `updateComponent` method throws an exception and the request returns `500`, so I think it's necessary to restrict the field `id` there and simply ignore it. 

I don't know whether there should be more restrictions, I think it's more a strategic question than a technical one if it's necessary to update for example the name of a component.